### PR TITLE
CC-1791: Added compression options to raw value example

### PR DESCRIPTION
**Pull request by rhauch**
This only changes the documentation.

### CODE REVIEW

File: kafka-connect-s3/docs/s3_connector.rst [hunk 1]
```diff
@@ -314,3 +314,87 @@ schema generator class to its default value.
 Finally, schema evolution is disabled in this example by setting ``schema.compatibility`` to ``NONE``, as explained above.
 
 
+Write raw message values into S3
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+It is possible to use the S3 connector to write out the unmodified original message values into
+newline-separated files in S3. We accomplish this by telling Connect to not deserialize any of the
```
Comment (sansanichfb) [inline kafka-connect-s3/docs/s3_connector.rst]: I am wondering whether message itself can contain escaped newline character?

Comment (jklukas) [reply kafka-connect-s3/docs/s3_connector.rst]: I agree that's worth bringing up here. [The docs for the FORMAT_BYTEARRAY_LINE_SEPARATOR_CONFIG do specify](https://github.com/confluentinc/kafka-connect-storage-cloud/blob/master/kafka-connect-s3/src/main/java/io/confluent/connect/s3/S3SinkConnectorConfig.java#L318-L319):

> An input record that contains the line separator will look like multiple records in the output S3 object.

So, the connector won't break, but the output may not be interpreted the way you expect if it contains the separator, and users need to be aware of that responsibility.

File: kafka-connect-s3/docs/s3_connector.rst [hunk 2]
```diff
@@ -314,3 +314,87 @@ schema generator class to its default value.
...
+  s3.region=us-west-2
+  s3.part.size=5242880
+  s3.compression.type=gzip
+
+The ``s3.bucket.name`` is mandatory and names your S3 bucket where the exported Kafka records
+should be written. Other useful settings are ``s3.region``, which you should set if you use a
+region other than the default, and ``s3.part.size`` to control the size of each part in the
+multipart uploads that will be used to upload a single chunk of Kafka records.
+The ``s3.compression.type`` specifies that we want the S3 connector to compress our S3 objects
+using GZIP compression, adding the ``.gzip`` extension to any files (see below).
```
Comment (jklukas) [inline kafka-connect-s3/docs/s3_connector.rst]: `gzip` is the correct name for the `s3.compression.type` value, but the extension added to files is `.gz` ([second argument to CompressionType](https://github.com/confluentinc/kafka-connect-storage-cloud/blob/v4.1.0-rc3/kafka-connect-s3/src/main/java/io/confluent/connect/s3/storage/CompressionType.java#L39)).

File: kafka-connect-s3/docs/s3_connector.rst [hunk 3]
```diff
@@ -314,3 +314,87 @@ schema generator class to its default value.
...
+
+  partitioner.class=io.confluent.connect.storage.partitioner.TimeBasedPartitioner
+  timestamp.extract=Wallclock
+
+Custom partitioners are always an option, too. Just be aware that since the record value is
+an opaque binary value, we cannot extract timestamps from fields using the ``RecordField``
+option.
+
+The S3 connector configuration outlined above results in newline-delimited gzipped objects in S3
+with ``.bin.gzip``.
```
Comment (jklukas) [inline kafka-connect-s3/docs/s3_connector.rst]: Should be `.bin.gz` here.

File: kafka-connect-s3/docs/s3_connector.rst (hunk 1 above)
Comment (jklukas) [reply kafka-connect-s3/docs/s3_connector.rst]: It's probably appropriate to discuss this detail later in the example, when we mention `format.bytearray.separator`.

**Comment by rhauch** [comment]
@jklukas thanks for the review. I fixed the `.gz` extension in both places. Ready for second pass.

**Comment by rhauch** [comment]
Rebased after #158 was merged.

**Comment by rhauch** [comment]
@kkonstantine I adjusted the text to include this statement:

> but you can control this with the ``format.bytearray.separator`` setting. 

to say

> but you can control this with the ``format.bytearray.separator`` setting, and you may want to consider this if your messages might contain newlines.

**Comment by kkonstantine** [comment]
To the point. Thanks @rhauch ! LGTM

**Comment by rhauch** [comment]
Waiting to merge until we have a `4.1.0-post` branch. This PR should then be changed to use that branch rather than `4.1.x` so that it goes into the 4.1.0 documentation.

**Comment by kkonstantine** [comment]
@rhauch this is now ready. Do you mind targeting `4.1.0-post` and rebasing?

**Pull request merged**